### PR TITLE
Allow for static groups

### PR DESCRIPTION
**Pull request by nickshine**
This adds the ability to include pre-existing Artifactory groups in a generated token, independent of the dynamically-generated group and permissions.

### Before:
```sh
➜  vault write artifactory-cloud/roles/role1 token_ttl=600 permission_targets=@scripts/sample_permission_targets.json
Key                   Value
---                   -----
permission_targets    [
  {
    "repo": {
      "include_patterns": ["@my-scoped/my-scoped-package/**"],
      "exclude_patterns": [""],
      "repositories": ["npm-test-west-local", "npm-test-east-local"],
      "operations": ["read", "write", "annotate"]
    }
  },
  ...
]
role_id               07ffa23817eb0c999f8e126a39db481e
role_name             role1

➜  vault write artifactory-cloud/token/role1 ttl=60
Key             Value
---             -----
access_token    eyJ2Z...
username        auto-vault-plugin.role1
```
Acess token payload (before):
```json
{
  "sub": "jfac@01hq2njf0pvbcv0g0m368z0s0z/users/auto-vault-plugin.role1",
  "scp": "applied-permissions/groups:vault-plugin.07ffa23817eb0c999f8e126a39db481e",
  "aud": "*@*",
  "iss": "jfac@01hq2njf0pvbcv0g0m368z0s0z",
  "exp": 1708580644,
  "iat": 1708580584,
  "jti": "15948195-00ae-4eb9-a9fa-038d4f92e48f"
}
```

### After

With static groups and permission targets:
```sh
➜  vault write artifactory-cloud/roles/role1 token_ttl=600 permission_targets=@scripts/sample_permission_targets.json groups=group1,group2
Key                   Value
---                   -----
groups                [group1 group2]
permission_targets    [
  {
    "repo": {
      "include_patterns": ["@my-scoped/my-scoped-package/**"],
      "exclude_patterns": [""],
      "repositories": ["npm-test-west-local", "npm-test-east-local"],
      "operations": ["read", "write", "annotate"]
    }
  },
 ...
]
role_id               07ffa23817eb0c999f8e126a39db481e
role_name             role1

```

Access token payload:

```json
{
  "sub": "jfac@01hq2njf0pvbcv0g0m368z0s0z/users/auto-vault-plugin.role1",
  "scp": "applied-permissions/groups:vault-plugin.07ffa23817eb0c999f8e126a39db481e,group1,group2",
  "aud": "*@*",
  "iss": "jfac@01hq2njf0pvbcv0g0m368z0s0z",
  "exp": 1708580899,
  "iat": 1708580839,
  "jti": "568b29a4-e391-4fb0-a2f3-268f603be99f"
}

```

With static groups only:

```
➜  vault write artifactory-cloud/roles/role2 token_ttl=600 groups=group1,group2                                                                                                    
Key                   Value
---                   -----
groups                [group1 group2]
permission_targets    n/a
role_id               e928e822a330518bb944af5aeca8cf99
role_name             role2
```

Access token payload:

```json
{
  "sub": "jfac@01hq2njf0pvbcv0g0m368z0s0z/users/auto-vault-plugin.role2",
  "scp": "applied-permissions/groups:group1,group2",
  "aud": "*@*",
  "iss": "jfac@01hq2njf0pvbcv0g0m368z0s0z",
  "exp": 1708581046,
  "iat": 1708580986,
  "jti": "54b84818-9734-43c0-b6e6-d452696cce35"
}

```

### CODE REVIEW

**Review by lootek — COMMENTED** [review]
LGTM, nice work! 👍

**Review by SingulariteeSplunk — APPROVED** [review]
LGTM 👍

File: plugin/artifactory_client.go [hunk 1]
```diff
@@ -185,9 +186,15 @@ func (ac *artifactoryClient) DeletePermissionTarget(ptName string) error {
...
+	var groups []string
+	if len(role.PermissionTargets) > 0 {
+		groups = append(groups, groupName(role))
+	}
+	groups = append(groups, role.Groups...)
+
 	params := accessservices.CreateTokenParams{
 		CommonTokenParams: auth.CommonTokenParams{
-			Scope:     fmt.Sprintf("applied-permissions/groups:%s", groupName(role)),
+			Scope:     fmt.Sprintf("applied-permissions/groups:%s", strings.Join(groups, ",")),
```
Comment (lootek) [inline plugin/artifactory_client.go:197]: Is this guaranteed the comma is disallowed in the group name? If there's no such validation before we can get into trouble here

**Review by lootek — APPROVED** [review]

File: plugin/artifactory_client.go (hunk 1 above)
Comment (nickshine) [reply plugin/artifactory_client.go:197]: There is no validation but we generate the role name and there is no way to override it from a consumer perspective.

**Pull request merged**